plugins/essif-lab/src/Application/Workflows/ManageHooks.php

```diff
@@ -39,43 +50,35 @@ private function requestIsEmpty($request): bool {
 
 	private function doesHookAlreadyExists($hooks, $request): bool {
 		return count(array_filter($hooks, function ($v) use ($request) {
-				return $v['context'] !== $request['context'] || $v['target'] !== $request['target'];
+				return $v[self::CONTEXT] !== $request[self::CONTEXT] || $v[self::TARGET] !== $request[self::TARGET];
 			})) !== count($hooks);
 	}
 
-	private function generateID($hooks): int {
-		$last = is_array($hooks) && array_key_exists(count($hooks) - 1, $hooks) ? $hooks[count($hooks) - 1] : null;
-
-		return ! empty($last) && array_key_exists('ID', $last) ? $last['ID'] : 1;
+	private function generateHookRecord($hooks, $request): array {
+	    $request["ID"] = $this->generateID($hooks);
+		return $request;
 	}
 
-	private function generateHookRecord($hooks, $request): array {
-		return array_merge(['ID' => $this->generateID($hooks)], $request);
+	private function generateID($hooks): int {
+        return is_array($hooks) && array_key_exists(self::MAX_ID, $hooks) ? ++$hooks[self::MAX_ID] : 1;
```

Comment:
In principe wat je wil, is een sequence bijhouden, dus is het niet het max ID maar last ID. De check zou dus als volgt moeten zijn. Als er nog geen last ID attribuut in de array zit is de waarde 1. Als het attribuut wel bestaat dan wordt die increment uitgevoerd. In beide gevallen moet de uiteindelijke waarde aan de hooks array worden toegevoegd.